src/stages/com.famedly.login.sso/openid.ts

```
@@ -163,7 +166,21 @@ export class OidcProvider {
 		private id: string,
 		/** The relying party oidc callback url */
 		private oidcCallbackUrl: string,
-	) { }
+	) {
+		// Use HTTP request timeout from config if provided correctly, otherwise use default
+		if (config.timeout_ms) {
+			if (Number.isInteger(config.timeout_ms) && config.timeout_ms > 0) {
+				log.verbose(`HTTP request timeout for provider '${id}' configured as ${config.timeout_ms} ms.`);
+				this.timeoutMs = config.timeout_ms;
+			} else {
```

Personally, I think the program should simply fail to start if a value has been provided but it's incorrect, rather than falling back to a default. The user intent is clearly to change the value, and not succeeding in this should fail in a way the user is sure to notice.

That said, I also don't think it's *that* important, especially for this component right now, so if you would prefer to just keep it like this that's also fine.

---

I completely agree! It was also my first approach, just according to the "fail fast and fail safe" principle. But then I had doubts about whether it would be a good idea to introduce a whole new "error type" because of this one small change. But if you also think it's right, I'll be happy to change it that way.